File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // Brute force Approach

        // int n = nums.size();
        // for(int i = 0;i<n;i++){
        //     for(int j = i+1; j<n;j++){
        //         if(nums[i]+nums[j] == target){
        //             return {i,j};
        //         }
        //     }
        // }
        // return {-1,-1};

        // Hashing Approach

        unordered_map<int,int>mpp;
        for(int i =0; i<nums.size(); i++){
            int more = target - nums[i];
            if(mpp.find(more) != mpp.end()){
                return {mpp[more],i};
            }
            mpp[nums[i]] = i;
        }
        return {-1,-1};

        // Optimal Approach 

        // int i =0;
        // int j = nums.size()-1;
        // sort(nums.begin(),nums.end());

    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Brute
        # for i in range(0, len(nums)-1):
        #     for j in range(i+1, len(nums)):
        #         if nums[i]+nums[j] == target:
        #             return [i,j]

        # Optimal Solution
        num_map = {}  # Hash table to store number and its index
        for i in range(0, len(nums)):
            remaining = target - nums[i]  # Find the remaining
            if remaining in num_map:
                return [num_map[remaining], i] 
            num_map[nums[i]] = i  
